### PR TITLE
testing/php7-solr: modernize

### PR DESCRIPTION
**Pull request by TBK**
Made a mistake when cherrypicking for https://github.com/alpinelinux/aports/pull/3784, so it did not make it.

### CODE REVIEW

File: testing/php7-solr/APKBUILD [hunk 1]
```diff
@@ -4,33 +4,37 @@ pkgname=php7-solr
 _pkgreal=solr
 pkgver=2.4.0
 _pkgver=${pkgver}
-pkgrel=1
+pkgrel=2
 pkgdesc="Apache SOLR extension for PHP"
 url="http://pecl.php.net/package/$_pkgreal"
 arch="all"
 license="PHP"
+options="!check" # As of 2.4.0 the test fails
```
Comment (andypost) [inline testing/php7-solr/APKBUILD]: Any idea why test fails? I gonna try it on aarch64

Comment (andypost) [reply testing/php7-solr/APKBUILD]: dependency is wrong
```
aports/testing/php7-solr/src/solr-2.4.0/tests$ cat 000.solr_server_compat.log 

---- EXPECTED OUTPUT
bool(true)
---- ACTUAL OUTPUT
PHP Warning:  PHP Startup: Unable to load dynamic library '/mnt/testing/php7-solr/src/solr-2.4.0/modules/solr.so' - Error relocating /mnt/testing/php7-solr/src/solr-2.4.0/modules/solr.so: php_json_decode_ex: symbol not found in Unknown on line 0

Warning: PHP Startup: Unable to load dynamic library '/mnt/testing/php7-solr/src/solr-2.4.0/modules/solr.so' - Error relocating /mnt/testing/php7-solr/src/solr-2.4.0/modules/solr.so: php_json_decode_ex: symbol not found in Unknown on line 0

Fatal error: Uncaught Error: Class 'SolrClient' not found in /mnt/testing/php7-solr/src/solr-2.4.0/tests/000.solr_server_compat.php:14
Stack trace:
#0 {main}
  thrown in /mnt/testing/php7-solr/src/solr-2.4.0/tests/000.solr_server_compat.php on line 14
---- FAILED
```

Comment (TBK) [reply testing/php7-solr/APKBUILD]: php7-json (json.so) is not loaded: the "php_json_decode_ex" method is from this file - https://github.com/php/php-src/blob/1314f0fb3a9ff25b9afe68845012c1d0acbec262/ext/json/json.c

Comment (andypost) [reply testing/php7-solr/APKBUILD]: I used to hack Makefile and run tests
```
$(PHP_EXECUTABLE) -n -c $(top_builddir)/tmp-php.ini $(PHP_TEST_SETTINGS) $(top_srcdir)/run-tests.php -n -c $(top_builddir)/tmp-php.ini -d extension=/usr/lib/php7/modules/json.so ....
```

All failed tests require Solr installed & running so please update comment about that

```
=====================================================================
TEST RESULT SUMMARY
---------------------------------------------------------------------
Exts skipped    :    0
Exts tested     :   13
---------------------------------------------------------------------

Number of tests :   85                85
Tests skipped   :    0 (  0.0%) --------
Tests warned    :    0 (  0.0%) (  0.0%)
Tests failed    :   30 ( 35.3%) ( 35.3%)
Expected fail   :    0 (  0.0%) (  0.0%)
Tests passed    :   55 ( 64.7%) ( 64.7%)
---------------------------------------------------------------------
Time taken      :    4 seconds
=====================================================================

=====================================================================
FAILED TEST SUMMARY
---------------------------------------------------------------------
Solr Server 4+ Compatibility Test [tests/000.solr_server_compat.phpt]
SolrClient::ping() - Ping Solr Server [tests/000.solrclient_ping.phpt]
SolrClient::addDocument() - sending document to the Solr server [tests/001.solrclient_addDocument.phpt]
SolrClient::addDocuments() - sending multiple documents to the Solr server [tests/002.solrclient_addDocuments.phpt]
SolrClient::addDocument() - Overwrite an existing document [tests/003.solrclient_addDocument_no_overwrite.phpt]
SolrClient::query() - Sending a chained query request [tests/004.solrclient_query_response.phpt]
SolrClient::query() - throw SolrClientException on Solr Server exception [tests/005.solrclient_query_error.phpt]
SolrClient::system() - Retrieve Solr Server/System Info [tests/006.solrclient_system.phpt]
SolrClient::deleteByIds() - Test deleting documents by an array of IDS [tests/007.solrclient_deleteByIds.phpt]
SolrClient::commit() - Testing commit [tests/008.solrclient_commit.phpt]
SolrClient::optimize() - Testing optimize [tests/009.solrclient_optimize.phpt]
SolrClient::getById() - Testing Real Time Get by id [tests/012.solrclient_getById.phpt]
SolrClient::getByIds() - Testing Real Time Get by ids [tests/013.solrclient_getByIds.phpt]
SolrClient::addDocument() - Sending a document with child documents [tests/014.solrclient_addDocument_nested_document.phpt]
SolrClient::addDocuments() - Sending multiple documents with child documents [tests/015.solrclient_addDocuments_nested_document.phpt]
SolrClient::query() - Query child documents SolrObject (Integ) [tests/024.solrdocument_child_fetch.phpt]
SolrQuery - testing HighlightingParameters [tests/063.solrquery_HighlightingParameters.phpt]
SolrServerException (xml) - Throw Exception on solr server exception using xml response writer [tests/090.solrserverexception_xml.phpt]
SolrServerException (json) - Throw Exception on solr server exception using json response writer [tests/091.solrserverexception_json.phpt]
SolrServerException (phps) - Throw Exception on solr server exception using phps response writer [tests/092.solrserverexception_php.phpt]
SolrResponse::getResponse - JSON response writer [tests/100.solrresponse_json.phpt]
SolrResponse - ParseMode [tests/101.solrresponse_parseMode.phpt]
SolrResponse::getResponse - PHPS response writer [tests/102.solrresponse_phps.phpt]
SolrResponse::getArrayResponse - For All response writers [tests/103.solrresponse_get_array_response.phpt]
SolrResponse::getResponse() - maxScore [tests/104.solrresponse_get_response_maxscore.phpt]
Solr - Fetch and Update nested documents [tests/160.solr_update_document_block.phpt]
Solr Bug #69156 - segfault on 500 response from Solr [tests/bug_69156.phpt]
Solr Bug #70482 - Segmentation fault on server 500 response with no msg field XML RW [tests/bug_70482.phpt]
Solr Bug #70495 - Failed to parse 500 error response when Solr it lacks msg JSON RW [tests/bug_70495.phpt]
Solr Bug #70496 - Failed to parse 500 error response when Solr it lacks msg PHPS RW [tests/bug_70496.phpt]
=====================================================================

```

```
Fatal error: Uncaught SolrClientException: Solr HTTP Error 6: 'Couldn't resolve host name'  in /mnt/testing/php7-solr/src/solr-2.4.0/tests/000.solr_server_compat.php:16
```

Comment (TBK) [reply testing/php7-solr/APKBUILD]: @andypost Done.

**Pull request closed without merging**